pkg/ccl/backupccl/backup_test.go

```diff
@@ -9214,3 +9215,267 @@ func TestBackupRestoreSeparateIncrementalPrefix(t *testing.T) {
+					JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(), // speeds up test
+				},
+			},
+		})
+	defer cleanup()
+
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.rangefeed.enabled = true`)
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`)
+
+	conn := tc.ServerConn(0)
+	waitForTableSplit := func(tableName string) {
+		testutils.SucceedsSoon(t, func() error {
+			count := 0
+			if err := conn.QueryRow(
+				"SELECT count(*) "+
+					"FROM crdb_internal.ranges_no_leases "+
+					"WHERE table_name = $1 "+
+					"AND database_name = current_database()",
+				tableName).Scan(&count); err != nil {
+				return err
+			}
+			if count == 0 {
+				return errors.New("waiting for table split")
+			}
+			return nil
+		})
+	}
+
+	getTableStartKey := func() roachpb.Key {
+		row := conn.QueryRow(
+			"SELECT start_key "+
+				"FROM crdb_internal.ranges_no_leases "+
+				"WHERE table_name = $1 "+
+				"AND database_name = current_database() "+
+				"ORDER BY start_key ASC "+
+				"LIMIT 1",
+			"foo")
+
+		var startKey roachpb.Key
+		require.NoError(t, row.Scan(&startKey))
+		return startKey
+	}
+
+	getStoreAndReplica := func() (*kvserver.Store, *kvserver.Replica) {
+		startKey := getTableStartKey()
+		// Okay great now we have a key and can go find replicas and stores and what not.
+		r := tc.LookupRangeOrFatal(t, startKey)
+		l, _, err := tc.FindRangeLease(r, nil)
+		require.NoError(t, err)
+
+		lhServer := tc.Server(int(l.Replica.NodeID) - 1)
+		return getFirstStoreReplica(t, lhServer, startKey)
+	}
+
+	waitForExcludeDataFromBackup := func() {
+		testutils.SucceedsSoon(t, func() error {
+			_, r := getStoreAndReplica()
+			if !r.ExcludeReplicaFromBackup() {
+				return errors.New("waiting for exclude_data_from_backup to be applied")
+			}
+			return nil
+		})
+	}
+
+	sqlDB.Exec(t, `CREATE TABLE data.foo (id INT, INDEX bar(id))`)
+	sqlDB.Exec(t, `INSERT INTO data.foo select * from generate_series(1,10)`)
+
+	// Create another table.
+	sqlDB.Exec(t, `CREATE TABLE data.bar (id INT, INDEX bar(id))`)
+	sqlDB.Exec(t, `INSERT INTO data.bar select * from generate_series(1,10)`)
+
+	// Set foo to exclude_data_from_backup and back it up. The ExportRequest
+	// should be a noop and backup no data.
+	sqlDB.Exec(t, `ALTER TABLE data.foo SET (exclude_data_from_backup = true)`)
+	waitForTableSplit("foo")
+	waitForTableSplit("bar")
+	waitForExcludeDataFromBackup()
+	sqlDB.Exec(t, `BACKUP DATABASE data TO $1`, LocalFoo)
+
+	restoreDB.Exec(t, `RESTORE DATABASE data FROM $1`, LocalFoo)
+	require.Len(t, restoreDB.QueryStr(t, `SELECT * FROM data.foo`), 0)
+	require.Len(t, restoreDB.QueryStr(t, `SELECT * FROM data.bar`), 10)
+}
+
+// TestExportRequestBelowGCThresholdOnDataExcludedFromBackup tests that a
+// `BatchTimestampBeforeGCError` on an ExportRequest targeting a table that has
+// been marked as excluded from backup, does not cause the backup to fail.
+func TestExportRequestBelowGCThresholdOnDataExcludedFromBackup(t *testing.T) {
```

Comment:
I really wish there were a better way to write these tests that didn't need to have just so much boilerplate to get the environment just right to exercise that one behavior. I don't have a constructive alternative suggestion here so carry on by all means, just lamenting/hoping someone smarter than me spots a way to turn tests like this into datadriven with clever crdb_internal builtins or something.